sw/host/provisioning/orchestrator/src/ot_dut.py

```diff
@@ -0,0 +1,181 @@
+            host_flags = host_flags.format("teacup")
+            host_flags += " --disable-dft-on-reset"
+            device_elf = device_elf.format("silicon_creator")
+
+        # Assemble CP command.
+        cmd = f"""{_CP_HOST_BIN} \
+        --rcfile= \
+        --logging=info \
+        {host_flags} \
+        --elf={device_elf} \
+        --device-id="{self.device_id}" \
+        --test-unlock-token="{format_hex(self.test_unlock_token, width=32)}" \
+        --test-exit-token="{format_hex(self.test_exit_token, width=32)}" \
+        --manuf-state="{_ZERO_256BIT_HEXSTR}" \
+        --wafer-auth-secret="{_ZERO_256BIT_HEXSTR}" \
+        """
+
+        # TODO: capture DIN portion of device ID and update device ID.
+
+        # Get user confirmation before running command.
+        logging.info(f"Running command: {cmd}")
+        if self.require_confirmation:
+            confirm()
+
+        # Run provisioning flow and collect logs.
+        res = run(cmd, f"{self.log_dir}/cp_out.log.txt",
+                  f"{self.log_dir}/cp_err.log.txt")
+        if res.returncode != 0:
+            logging.warning(f"CP failed with exit code: {res.returncode}.")
+            confirm()
+        else:
+            logging.info("CP completed successfully.")
+
+    def run_ft(self) -> None:
+        """Runs the FT provisioning flow on the target DUT."""
+        logging.info("Running FT provisioning ...")
+
+        # Set cmd args and device ELF.
+        host_bin = _FT_HOST_BIN.format(self.sku_config.name)
+        host_flags = _BASE_PROVISIONING_FLAGS
+        individ_elf = _FT_INDIVID_DEVICE_ELF
+        perso_bin = _FT_PERSO_DEVICE_BIN
+        fw_bundle_bin = _FT_FW_BUNDLE_BIN
+        if self.fpga:
+            # Set host flags and device binaries for FPGA DUT.
+            # No need to load another bitstream, we will take over where CP
+            # stage above left off.
+            host_flags = host_flags.format(self.fpga)
+            individ_elf = individ_elf.format(
+                self.sku_config.name, f"fpga_{self.fpga}_rom_with_fake_keys")
+            perso_bin = perso_bin.format(
+                self.sku_config.name, f"fpga_{self.fpga}_rom_with_fake_keys")
+            fw_bundle_bin = fw_bundle_bin.format(
+                self.sku_config.name, f"fpga_{self.fpga}_rom_with_fake_keys")
+        else:
+            # Set host flags and device binaries for Silicon DUT.
+            host_flags = host_flags.format("teacup")
+            host_flags += " --disable-dft-on-reset"
+            individ_elf = individ_elf.format(self.sku_config.name,
+                                             "silicon_creator")
+            perso_bin = perso_bin.format(self.sku_config.name,
+                                         "silicon_creator")
+            fw_bundle_bin = fw_bundle_bin.format(self.sku_config.name,
+                                                 "silicon_creator")
+
+        # Assemble bazel command.
```

Comment:
Is this actually assembling a bazel command, or is this assembling a command line for the host `ft` binary?

Reply:
ah good catch. just a command. will fix.

Reply:
Fixed in #25167.